File: src/main/java/at/pria/joust/webapp/web/MainController.java
package at.pria.joust.webapp.web;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

@Controller
public class MainController {
    @RequestMapping(value = "/", method = RequestMethod.GET)
    public String index(Model model) {
        //TODO make an "upcoming matches" page
        return "redirect:/teams/";
    }

    @RequestMapping(value = "/error", method = RequestMethod.GET)
    public String error(Model model) {
        return "error";
    }
}

File: src/main/scala/at/pria/joust/webapp/web/MainController.scala
package at.pria.joust.webapp.web

import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod

@Controller
class MainController {
  @RequestMapping(value = Array("/error"), method = Array(RequestMethod.GET))
  def error(model: Model) = "error"
}
